File: HandSock-Server/src/main/java/com/server/handsock/websocket/handler/ClientHandler.java
package com.server.handsock.websocket.handler;

import com.corundumstudio.socketio.AckRequest;
import com.corundumstudio.socketio.SocketIOClient;
import com.server.handsock.sockets.service.OP_UserService;
import com.server.handsock.console.HandleResults;
import com.server.handsock.service.ClientService;

import java.util.HashMap;
import java.util.Map;

public class ClientHandler {

    private final Map<String, Object> service;
    private final HandleResults handleResults = new HandleResults();

    public ClientHandler(Map<String, Object> service) {
        this.service = service;
    }

    public void handleClientInit(SocketIOClient client, Map<String, Object> data, AckRequest ackRequest) {
        ClientService clientService = new ClientService(service);
        try {
            if (clientService.validClientToken(client)) {
                Map<String, Object> result = new HashMap<>();
                OP_UserService OPUserService = (OP_UserService) service.get("userService");
                result.put("userinfo", OPUserService.queryUserInfo(clientService.getRemoteUID(client)));
                ackRequest.sendAckData(handleResults.handleResultByCode(200, result, "获取成功"));
            } else ackRequest.sendAckData(handleResults.handleResultByCode(403, null, "请重新登录"));
        } catch (Exception e) {
            clientService.handleException(e, ackRequest);
        }
    }

}

File: HandSock-Server/src/main/java/com/server/handsock/socket/handler/ClientHandler.kt
package com.server.handsock.socket.handler

import com.corundumstudio.socketio.AckRequest
import com.corundumstudio.socketio.SocketIOClient
import com.server.handsock.client.service.ClientUserService
import com.server.handsock.common.utils.HandUtils
import com.server.handsock.service.AuthService
import com.server.handsock.service.ClientService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ClientHandler @Autowired constructor(
    private val authService: AuthService,
    private val clientService: ClientService,
    private val clientUserService: ClientUserService
) {
    fun handleClientInit(client: SocketIOClient, ackRequest: AckRequest) {
        ackRequest.sendAckData(authService.validClientStatusBySocket(client) {
            HandUtils.handleResultByCode(200, clientUserService.queryUserInfo(clientService.getRemoteUID(client)), "获取成功")
        })
    }
}
